### PR TITLE
feat(argo-workflows): add support for Deployment annotations

### PR DESCRIPTION
**Pull request by snooyen**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Comment by mkilchhofer** [comment]
Related to #1151 but for applicationset istead of workflows

File: charts/argo-workflows/templates/controller/workflow-controller-deployment.yaml [hunk 1]
```diff
@@ -5,6 +5,10 @@ metadata:
   labels:
     {{- include "argo-workflows.labels" (dict "context" . "component" .Values.controller.name "name" .Values.controller.name) | nindent 4 }}
     app.kubernetes.io/version: {{ default .Chart.AppVersion .Values.controller.image.tag | quote }}
+    {{- with .Values.controller.deploymentAnnotations }}
+    annotations:
+      {{- toYaml . | nindent 6 }}
+    {{- end }}
```
Comment (mkilchhofer) [inline charts/argo-workflows/templates/controller/workflow-controller-deployment.yaml:11]: wrong indent
```suggestion
  {{- with .Values.controller.deploymentAnnotations }}
  annotations:
    {{- toYaml . | nindent 4 }}
  {{- end }}
```

File: charts/argo-workflows/templates/server/server-deployment.yaml [hunk 2]
```diff
@@ -6,6 +6,10 @@ metadata:
   labels:
     {{- include "argo-workflows.labels" (dict "context" . "component" .Values.server.name "name" .Values.server.name) | nindent 4 }}
     app.kubernetes.io/version: {{ default .Chart.AppVersion .Values.server.image.tag | quote }}
+    {{- with .Values.server.deploymentAnnotations }}
+    annotations:
+      {{- toYaml . | nindent 6 }}
+    {{- end }}
```
Comment (mkilchhofer) [inline charts/argo-workflows/templates/server/server-deployment.yaml:12]: wrong indent
```suggestion
  {{- with .Values.server.deploymentAnnotations }}
  annotations:
    {{- toYaml . | nindent 4 }}
  {{- end }}
```

File: charts/argo-workflows/Chart.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: argo-workflows
 description: A Helm chart for Argo Workflows
 type: application
-version: 0.13.1
+version: 0.13.2
```
Comment (mkilchhofer) [inline charts/argo-workflows/Chart.yaml:5]: Let's bump minor version as this PR adds a feature.
```suggestion
version: 0.14.0
```

**Review by mkilchhofer — CHANGES_REQUESTED** [review]

**Comment by jmeridth** [comment]
@mkilchhofer can we please get a re-review of this?  We need this in order to use https://github.com/stakater/Reloader.  Thank you @snooyen for getting this PR written up.  Cheers.

**Comment by jmeridth** [comment]
@mkilchhofer bump

**Comment by jmeridth** [comment]
@mkilchhofer are you looking for more changes on this PR?

**Review by mkilchhofer — APPROVED** [review]

**Pull request merged**